app/src/androidTest/java/com/github/sdp_begreen/begreen/ParcelableDateTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.sdp_begreen.begreen
+
+import android.os.Parcel
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import java.util.*
+
+class ParcelableDateTest {
+    @Test
+    fun parcelableDateWriteToParcelWorks() {
+        val date = ParcelableDate(Date())
+        val parcel = Parcel.obtain()
+        date.writeToParcel(parcel, 0)
+        parcel.setDataPosition(0)
+        val createdFromParcel = ParcelableDate.CREATOR.createFromParcel(parcel)
+        assertThat(date.date?.time, equalTo(createdFromParcel.date?.time))
+    }
+
+    @Test
+    fun parcelableDateDescribeContentsWorks() {
+        val date = ParcelableDate(Date())
+        assertThat(date.describeContents(), equalTo(0))
+    }
+
+    @Test
+    fun parcelableDateNewArrayWorks() {
+        val date = ParcelableDate(Date())
+        val array = ParcelableDate.CREATOR.newArray(1)
+        assertThat(array.size, equalTo(1))
+    }
+
+    @Test
+    fun parelableDateConstructorWorks() {
+        val date = Date()
+        val parcel = Parcel.obtain()
+        parcel.writeLong(date.time)
+        parcel.setDataPosition(0)
+        val createdFromParcel = ParcelableDate(parcel)
+        assertThat(date.time, equalTo(createdFromParcel.date?.time))
+    }
+
+    @Test
+    fun parcelableDateCreatorWorks() {
```

Comment:
Tests seems exhaustive! To make them perfect you can just refactor the names by expliciting the expected behavior instead of "Works" but its not necessary

Reply:
Yes it's indeed a good idea